agent/testagent.go

```diff
@@ -81,6 +82,9 @@ type TestAgent struct {
 	// Agent is the embedded consul agent.
 	// It is valid after Start().
 	*Agent
+
+	// NoWaitForStartup, if false, will start the anti entropy sync and waitForUp() (see *waitForUp()).
+	NoWaitForStartup bool
```

Comment:
I don't think I understand the need for this. Why does `StartSync` and `waitForUp` cause a problem for this test?

Reply:
So the unit test that's here test the first scenario:

>* server startup

By the time `waitForUp` finishes:

> // waitForUp waits for leader election, or waits for the agent HTTP
> // endpoint to start responding, depending on the agent config.

that state will have passed already. So the metric will emit `1` since it's a healthy cluster.

---

I could probably leave `StartSync()` outside of the if block but I'm not sure it'd be useful for folks since I don't see a scenario for an agent in that state.

---

Does this make some sense ? 🤔